### PR TITLE
can config es id's constituent part

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
default the es's id use mysql's PK value, and now can config the id's constituent part use mysql's other column.

### CODE REVIEW

File: river/rule.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ type Rule struct {
 	Index  string `toml:"index"`
 	Type   string `toml:"type"`
 	Parent string `toml:"parent"`
+	Id []string `toml:"id"`
```
Comment (siddontang) [inline river/rule.go]: g/Id/ID/s

File: river/sync.go [hunk 2]
```diff
@@ -384,20 +384,33 @@ func (r *River) makeUpdateReqData(req *elastic.BulkRequest, rule *Rule,
...
 
-// Get primary keys in one row and format them into a string
-// PK must not be nil
+// If Id in toml file is none, get primary keys in one row and format them into a string, and PK must not be nil
+// Else get the Id's column in one row and format them into a string
 func (r *River) getDocID(rule *Rule, row []interface{}) (string, error) {
-	pks, err := canal.GetPKValues(rule.TableInfo, row)
-	if err != nil {
-		return "", err
+	var id []interface{}
```
Comment (siddontang) [inline river/sync.go]:
```
var (
  ids []interface{}
  err error 
)
```

**Comment by siddontang** [comment]
Please add test for this feature.

**Comment by WangXiangUSTC** [comment]
I have changed Id to ID, and add test, but can't pass the check.
could you tell me how to run test file in my computer? Or you can modify the test file.

File: etc/river.toml [hunk 3]
```diff
@@ -58,6 +58,8 @@ schema = "test"
 table = "t"
 index = "test"
 type = "t"
+# Default will use the mysql's primary key as es's id, if set id will use the id's column value as id
+id = ["id", "tags"]
```
Comment (siddontang) [inline etc/river.toml]: please use another rule section for this feature.
Btw, please add this to ReadMe too.

**Comment by siddontang** [comment]
Rest LGTM

**Comment by siddontang** [comment]
CI failed, please fix it.

**Comment by WangXiangUSTC** [comment]
ok

**Comment by WangXiangUSTC** [comment]
I have already modify the readme and example config file, and ci check passed

File: vendor/github.com/siddontang/go-mysql/canal/rows.go [hunk 4]
```diff
@@ -53,6 +53,16 @@ func GetPKValues(table *schema.Table, row []interface{}) ([]interface{}, error)
 	return values, nil
 }
 
+// Get term column's value
+func GetColumnValue(table *schema.Table, column string, row []interface{}) (interface{}, error) {
```
Comment (siddontang) [inline vendor/github.com/siddontang/go-mysql/canal/rows.go]: please update go-mysql commit version in glide.yml

**Comment by WangXiangUSTC** [comment]
done

**Comment by siddontang** [comment]
Thanks @WangXiangUSTC

**Pull request merged**